src/Workspaces/Core/Portable/Workspace/Solution/SyntacticDocument.cs

```diff
@@ -30,5 +32,8 @@ protected SyntacticDocument(Document document, SourceText text, SyntaxTree tree,
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             return new SyntacticDocument(document, text, root.SyntaxTree, root);
         }
+
+        public TService GetLanguageService<TService>() where TService : class, ILanguageService
+            => this.Document.GetLanguageService<TService>();
```

Comment:
The benefit here is mostly around the confusion that i run into a lot when trying to use this 'extension' on a 'document' that turns out to actually be a SemanticDocument.  i.e. i do:

```c#
document.GetLanguageService<ISyntaxFactsService>();
```

it doesn't work.

i try to add using.  it doesn't work.  i think something is broken.  I then have to poke aaround and realize "oh right, it's this strange type called 'XXXDocument' which isn't actually a Document :)

I personally think it's worth it just for the annoyance factor.

Reply:
I personally think every time you have a SemanticDocument you should name it `semanticDocument` rather than `document` to avoid the confusion. It's still a different thing a you need to be aware of that.

Reply:
I prefer this solution.

Reply:
I guess I just don't see this as a sustainable approach. Are we going to copy all methods and extensions that `Document` has and forward them?

Reply:
I guess I just don't see this as a sustainable approach. Are we going to copy all methods and extensions that `Document` has and forward them?

Reply:
If there are other extensions that we commonly run into problems with?  Sure. 

It's less an issue with non-extensions because, of course, intellisense just works.

Reply:
I'm with @Neme12: if your semanticDocuments are called document, fix that. :smile:

Reply:
Done.